### ISSUE TITLE
Allow to include multiple folders from separate locations

### ISSUE THREAD

**Issue by jotoeri** [nextcloud/notes#670]
**Is your feature request related to a problem? Please describe.**
Using the App for shared notes, as well as for private ones, there is still only one notes folder to share. Notes just get created there, independent of it's sharing status. Also when having a own folder-structure, it is not possible to use notes from separate folders.

**Describe the solution you'd like**
It would be cool, if one could just add arbitrary folders to be read by the app, that then appear just as the current categories. Advanced version of this feature would then be to allow admins to add such folders for all users within settings.

**Describe alternatives you've considered**
- Packing all data within one Notes Folder?
- Creating separate folders within the notes folder, that then can be shared?

**Comment by stefan-niedermann** [comment]
Just an idea: Have you tried to mount different subfolders e. g. via `WebDAV` into the `Notes` folder?

**Comment by jotoeri** [comment]
Hm, thats one possible workaround, yes. But seems to be quite slow on opening and storing again. However that brought me to another workaround of a symlink on the server, which indeed requires root access, but is much faster than external storage. Thanks for this idea @stefan-niedermann 👍 

Still, manually selecting different folders within the settings would be a really cool thing and would in my opinion increase the flexibility of that app quite much! :)

**Comment by stefan-niedermann** [comment]
Yes, i have concerns regarding the maintenance of this feature though. Practically it would end up with each selected folder to be a roor category - there are various cases which aren't both, easy to maintain *and* easy to understand for users, for example "why can't i create a note without a category?". It would also be lots of effort to decide (or implement choices) what happens to the notes when switching between "single folder" mode and "multi folder mode"

From my side it is unfortunately a 👎  because it seems to be an edgecase with several working workarounds and the ratio of implementing, maintaining and understanding this feature isn't it worth.

Final decision is up to @korelstar of course.

**Comment by jotoeri** [comment]
> why can't i create a note without a category?

Hm, how about everything without category goes to the current default Notes folder?

> effort to decide what happens to the notes

Just keep the current Notes-Folder where it is and include it as atm?
-> In fact, one would just add new categories, having a different path than `Notes/Category`. Needs of course to store (and use) a specific path per category.

> several working workarounds

Depends on if you accept as workaround, what is slow (and strange - instance uses outside interface to request data from itself?!) or one, which requires ssh root access?

**Comment by korelstar** [comment]
This is related to #285, since your motivation is the same (managing shared notes). However, I think sharing categories from within the app (as discussed in #285) would be more advantageous than mapping categories to folders outside the notes directory after sharing them outside of the app. Hence, I agree with @stefan-niedermann , that your approach is rather complicated. I think, we should concentrate on implementing a sharing feature in the notes app itself.

For the meantime, there is a workaround:

> Another way would be to share a folder inside the notes folder to your group and let every user move that shared folder from their root directory to their notes folder. That's the way I'm using notes in order to have both: a shared folder and my own notes.

**Comment by stefan-niedermann** [comment]
> Hm, how about everything without category goes to the current default Notes folder?

What do you do when the user tries to create a subcategorie in the Notes folder with the same name as a mounted folder?

> Just keep the current Notes-Folder where it is and include it as atm?

How do you handle conflicts? Given a category ABC exists and the user wants to mount a folder ABC?

One dialog per scenario? Automatic decisions? How to teach the user whats happening?

There are so many pitfalls abd scenarions ... It ain't that easy as you mught imagine and the requeats for this exact issue seem to be quite little (you are the first - out of many thousand users) - so I still think the ratio of effort/complication and benefit is not enough at the moment.

**Comment by stefan-niedermann** [comment]
Let's focus on #285 🙂 🚀

**Comment by jotoeri** [comment]
> since your motivation is the same (managing shared notes).

Hm, actually sharing categories within the notes folder i do with groupfolders currently. That is quite easy to do.
Motivation was more to have project-notes within an existing folder-structure (sorted by project/sub-project, everything per project within respecitve folder), but for users to be able to use the notes app instead of the text-app for editing.

>  It ain't that easy as you mught imagine

Sorry, but that sounds to me like a high horse you are speking from towards me. 👎 Let's keep discussions at eye level.

Anyways. I see this issue is done. 😢

**Comment by stefan-niedermann** [comment]
I hit the wrong button, sorry 😄 Not my repository here, wasn't my intention to close it.

**Comment by stefan-niedermann** [comment]
> Sorry, but that sounds to me like a high horse you are speaking from towards me.

Also didn't want to sound rude, i just think that there are lots of edge cases and pitfalls which you might not have considered yet.

We can't just push a new feature in when there are so many questions left. And one of the questions (besides the [already mentioned once](https://github.com/nextcloud/notes/issues/670#issuecomment-766306247) is: Is this feature food for the majority of users or will it make things more complicated to satisfy a requirement which has never before been reported by anyone?

>  Let's keep discussions at eye level.

Well, we areasmall team and have to *maintain* and *test* this stuff, also in the future. This is not a one-shot for us and i want you to understand why we need discussions before adding new stuff. This has nothing to do with a "high horse" or "eye level".

But let's not get off topic. This is about your feature request, there are questiony left, i have spoken out my concerns and a decision have to be made.

**Comment by cdelledonne** [comment]
> However that brought me to another workaround of a symlink on the server, which indeed requires root access, but is much faster than external storage.

@jotoeri I just stumbled upon this issue. What is this workaround you have found? Symlink'ing a folder on the server doesn't seem to result in anything on the client side.

Thank you in advance :)

**Comment by jotoeri** [comment]
Hm, @cdelledonne,
what do you mean with the client? The Notes-App on the Browser or the notes android-client?
I think i that once symlinked a folder via ssh into the notes folder within the users data-storage. Might need to reread the folder-structure via occ to take effect on the notes app on the browser. About the android-client i don't know how it operates, so i also can't tell you if the workaround will take effect there... 🤷‍♂️ 🤔

**Comment by cdelledonne** [comment]
Ah, re-scanning the file database with `occ` worked indeed. Thanks!

**Comment by thiswillbeyourgithub** [comment]
Hi, on a related topic : can anyone tell me where I can access the actual note **files** on android ?

For markdown note taking on my android I'm currently using Markor ([here on f droid](https://gsantner.net/project/markor.html)) but would be interested in syncing them directly with nextcloud using Nextcloud Notes.

Is it possible? I basically only need to know the location of the files :)

Thanks!

**Comment by stefan-niedermann** [comment]
> Hi, on a related topic : can anyone tell me where I can access the actual note files on android ?

The [Notes Android app](https://github.com/stefan-niedermann/nextcloud-notes/) stores the notes in a local `.sqlite` database instead of files for performance reasons. You can use the [Nextcloud Android app](https://github.com/nextcloud/android) if you need access to the actual files.

**Comment by silopolis** [comment]
> actually sharing categories within the notes folder i do with groupfolders currently.

Hi, can you roughy explain how you do this please?

**Comment by Nawrux** [comment]
> What do you do when the user tries to create a subcategorie in the Notes folder with the same name as a mounted folder?

Only the mounted folder is used and if no mounted folder is set the default is being used

**Comment by Tinqerer** [comment]
Have others found a way to make this work? It'd be so good to have access to notes I keep organised in other folders, not just in the "Notes" folder.
